### PR TITLE
save and restore the maximized window state

### PR DESCRIPTION
**Pull request by feeva**
Maybe there was a reason this feature was not implemented. But if nothing matters, I think this is  a necessary feature.

### CODE REVIEW

**Comment by szwacz** [comment]
Yes, there was a reason. OSX and Linux (Ubuntu is the only I test) don't allow to restore windows as maximized. The only system where it is possible is windows (I assume this is the OS you have in mind with this PR). So your patch might have sense if some conditional only for windows will be applied.

File: app/helpers/window.js [hunk 1]
```diff
@@ -35,7 +35,8 @@ export default function (name, options) {
             x: position[0],
             y: position[1],
             width: size[0],
-            height: size[1]
+            height: size[1],
+			maximized: win.isMaximized()
```
Comment (szwacz) [inline app/helpers/window.js]: Something is wrong with the indentation of all your code.

File: app/helpers/window.js [hunk 2]
```diff
@@ -65,18 +66,28 @@ export default function (name, options) {
         }
         return windowState;
     };
+	
+	function updateState() {
```
Comment (szwacz) [inline app/helpers/window.js]: This project uses convention `var updateState = function () {` for declaring functions. Convention is convention :)

File: app/helpers/window.js [hunk 3]
```diff
@@ -65,18 +66,28 @@ export default function (name, options) {
...
-        }
+		updateState();
         userDataDir.write(stateStoreFile, state, { atomic: true });
     };
 
     state = ensureVisibleOnSomeDisplay(restore());
 
     win = new BrowserWindow(Object.assign({}, options, state));
-
+	if (state.maximized)
```
Comment (szwacz) [inline app/helpers/window.js]: Curly braces please :)

File: app/helpers/window.js (hunk 1 above)
Comment (feeva) [inline app/helpers/window.js]: It's odd as I myself prefer spaces to tabs and I setup up my VS Code as such.

Anyhow, your guess about my OS choice is right and I may test in other OSes and modify the code style if you are willing to accept this PR.

**Pull request closed without merging**